### PR TITLE
Bump `@metamask/gas-fee-controller` to `^10.0.1`

### PR DESCRIPTION
**Pull request by jiexi**
## **Description**

Bump `@metamask/gas-fee-controller` to `^10.0.1` to help get `@metamask/network-controller` versions aligned that way we have smaller bundle sizes in extension. Currently we have a few resolved versions, this helps reduce but not eliminate that.

please ignore the out-of-date branch name

## **Related issues**

See: https://github.com/MetaMask/metamask-extension/pull/21529

## **Manual testing steps**

Send some transactions. Gas fee estimate should work same as before

## **Screenshots/Recordings**

No user facing changes.

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by jiexi** [comment]
@metamaskbot update-policies

**Comment by jiexi** [comment]
@metamaskbot update-policies

**Comment by jiexi** [comment]
@metamaskbot update-policies

**Comment by jiexi** [comment]
@metamaskbot update-policies

**Comment by jiexi** [comment]
@metamaskbot update-policies

**Review by legobeat — DISMISSED** [review]
LGTM!

**Comment by jiexi** [comment]
@metamaskbot update-policies

**Review by legobeat — DISMISSED** [review]

File: yarn.lock [hunk 1]
```diff
@@ -4219,31 +4137,31 @@ __metadata:
   linkType: hard
 
 "@metamask/eth-sig-util@npm:^6.0.0":
-  version: 6.0.2
-  resolution: "@metamask/eth-sig-util@npm:6.0.2"
+  version: 6.0.0
+  resolution: "@metamask/eth-sig-util@npm:6.0.0"
```
Comment (adonesky1) [inline yarn.lock:4141]: This version is deprecated

File: yarn.lock [hunk 2]
```diff
@@ -4219,31 +4137,31 @@ __metadata:
...
-  checksum: e6072984d888389fe746834963be55c8b58c0698fcb5bb0a68d61ff885df9c2309a50b782fc0167a75438a2a3efd67321e5ce8a4a2a81b4d70ba07f40a875c09
+  checksum: ad257798831848f3fb82aba21d6ebc0cc15114a15a20bf288c785f722157646e71dc0dd6299d948b9c2c3d14fb56fa3c527d8c22fbb081640c87d967be2a0e75
   languageName: node
   linkType: hard
 
 "@metamask/eth-sig-util@npm:^7.0.0":
-  version: 7.0.1
-  resolution: "@metamask/eth-sig-util@npm:7.0.1"
+  version: 7.0.0
+  resolution: "@metamask/eth-sig-util@npm:7.0.0"
```
Comment (adonesky1) [inline yarn.lock:4155]: Also deprecated. We need this to resolve to 7.0.1

Comment (adonesky1) [reply yarn.lock:4155]: unclear to me why this change happened with this version bump...?

Comment (adonesky1) [reply yarn.lock:4155]: I think we to bump to atleast @metamask/gas-fee-controller@^10.0.1 and possible 11.0.0 to get the fix

Comment (jiexi) [reply yarn.lock:4155]: I don't see a 11.0.0 release

Comment (jiexi) [reply yarn.lock:4155]: bumped to 10.0.1 here https://github.com/MetaMask/metamask-extension/pull/21832/commits/ba19b6e8a0274e124ff3ff7e5b05beaa35221bd5

**Review by adonesky1 — APPROVED** [review]
LGTM!

**Review by shanejonas — APPROVED** [review]

**Pull request merged**